ddtrace/internal/ci_visibility/writer.py

```diff
@@ -81,3 +88,27 @@ def recreate(self):
             dogstatsd=self.dogstatsd,
             sync_mode=self._sync_mode,
         )
+
+    def _check_agent(self):
+        try:
+            info = agent.info()
+        except Exception:
+            info = None
+
+        if info:
+            endpoints = info.get("endpoints", [])
+            if endpoints and (EVP_PROXY_AGENT_BASE_PATH in endpoints or ("/" + EVP_PROXY_AGENT_BASE_PATH) in endpoints):
+                return self.STATE_AGENTPROXY
+        return self._state
+
+    def _set_state(self, new_state):
+        if new_state == self.STATE_AGENTPROXY:
+            self._state = new_state
+            self._endpoint = EVP_PROXY_AGENT_ENDPOINT
+            self.intake_url = agent.get_trace_url()
+            self._headers["X-Datadog-EVP-Subdomain"] = "citestcycle-intake"
+
+    def flush_queue(self, raise_exc=False):
+        if self._state != self.STATE_AGENTPROXY:
+            self._set_state(self._check_agent())
```

Comment:
This calls `_check_agent()` each time we call `flush_queue ()` which means doing an additional request. Should we cache the result that `agent.info()` returns for N seconds to avoid making lots of requests? Just an idea for an improvement, not a blocker, since it's a local request so it should be fast.

Reply:
Do we need to check the agent each flush? I wonder if switching from agentless to EVP proxy via the agent is something that should happen in the middle of a CI job.

Reply:
@Yun-Kim I don't know. @albertvaka do you have an answer about whether the agent check should happen once on startup or during execution?

Reply:
cc'ing @juan-fernandez here

Reply:
> Do we need to check the agent each flush?

no, we don't. We can do it once at the beginning of the test session and assume that the agent capabilities will stay fixed until the end of the test session. That's what other tracers are doing.

> I wonder if switching from agentless to EVP proxy via the agent is something that should happen in the middle of a CI job.

We can assume we are not prepared for that. We'll decide configuration at the beginning of the test session.